### PR TITLE
Adding documentation for APIs in System.Collections.Immutable targeted for 3.0

### PR DESCRIPTION
**Pull request by jozkee**
List of APIs documented:
* System.Collections.Immutable.ImmutableArray.ToImmutableArray``1(builder)
* System.Collections.Immutable.ImmutableDictionary.ToImmutableDictionary``2(builder)
* System.Collections.Immutable.ImmutableHashSet.ToImmutableHashSet``1(builder)
* System.Collections.Immutable.ImmutableList.ToImmutableList``1(builder)
* System.Collections.Immutable.ImmutableSortedDictionary.ToImmutableSortedDictionary``2(builder)
* System.Collections.Immutable.ImmutableSortedSet.ToImmutableSortedSet``1(builder)
* System.Collections.Immutable.ImmutableArray`1
  * AsMemory
  * AsSpan

### CODE REVIEW

**Comment by Jozkee** [comment]
@safern 
@mairaw  @rpetrusha @carlossanlop

File: xml/System.Collections.Immutable/ImmutableArray.xml [hunk 1]
```diff
@@ -901,10 +901,10 @@ This overload allows helper methods or custom builder classes to efficiently avo
         <Parameter Name="builder" Type="System.Collections.Immutable.ImmutableArray&lt;TSource&gt;+Builder" RefType="this" Index="0" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <typeparam name="TSource">To be added.</typeparam>
-        <param name="builder">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="TSource">The type of elements contained in the immutable array.</typeparam>
+        <param name="builder">The builder to create the immutable array from.</param>
+        <summary>Creates an immutable array of the current contents of the builder's array.</summary>
```
Comment (rpetrusha) [inline xml/System.Collections.Immutable/ImmutableArray.xml]:
```suggestion
        <summary>Creates an immutable array from the current contents of the builder's array.</summary>
```

File: xml/System.Collections.Immutable/ImmutableDictionary.xml [hunk 2]
```diff
@@ -531,11 +531,11 @@
...
       <Docs>
-        <typeparam name="TKey">To be added.</typeparam>
-        <typeparam name="TValue">To be added.</typeparam>
-        <param name="builder">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="TKey">The type of the keys in the dictionary.</typeparam>
+        <typeparam name="TValue">The type of the values in the dictionary.</typeparam>
+        <param name="builder">The builder to create the immutable dictionary from.</param>
+        <summary>Creates an immutable dictionary of the current contents of the builder's dictionary.</summary>
```
Comment (rpetrusha) [inline xml/System.Collections.Immutable/ImmutableDictionary.xml]:
```suggestion
        <summary>Creates an immutable dictionary from the current contents of the builder's dictionary.</summary>
```

File: xml/System.Collections.Immutable/ImmutableList.xml [hunk 3]
```diff
@@ -666,10 +666,10 @@
         <Parameter Name="builder" Type="System.Collections.Immutable.ImmutableList&lt;TSource&gt;+Builder" RefType="this" Index="0" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <typeparam name="TSource">To be added.</typeparam>
-        <param name="builder">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="TSource">The type of the elements in the list.</typeparam>
+        <param name="builder">The builder to create the immutable list from.</param>
+        <summary>Creates an immutable list of the current contents of the builder's collection.</summary>
```
Comment (rpetrusha) [inline xml/System.Collections.Immutable/ImmutableList.xml]:
```suggestion
        <summary>Creates an immutable list from the current contents of the builder's collection.</summary>
```

File: xml/System.Collections.Immutable/ImmutableHashSet.xml [hunk 4]
```diff
@@ -433,10 +433,10 @@
         <Parameter Name="builder" Type="System.Collections.Immutable.ImmutableHashSet&lt;TSource&gt;+Builder" RefType="this" Index="0" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <typeparam name="TSource">To be added.</typeparam>
-        <param name="builder">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="TSource">The type of the elements in the hash set.</typeparam>
+        <param name="builder">The builder to create the immutable hash set from.</param>
+        <summary>Creates an immutable hash set of the current contents of the builder's set.</summary>
```
Comment (rpetrusha) [inline xml/System.Collections.Immutable/ImmutableHashSet.xml]:
```suggestion
        <summary>Creates an immutable hash set from the current contents of the builder's set.</summary>
```

File: xml/System.Collections.Immutable/ImmutableSortedDictionary.xml [hunk 5]
```diff
@@ -412,11 +412,11 @@
...
       <Docs>
-        <typeparam name="TKey">To be added.</typeparam>
-        <typeparam name="TValue">To be added.</typeparam>
-        <param name="builder">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="TKey">The type of the keys in the dictionary.</typeparam>
+        <typeparam name="TValue">The type of the values in the dictionary.</typeparam>
+        <param name="builder">The builder to create the immutable sorted dictionary from.</param>
+        <summary>Creates an immutable sorted dictionary of the current contents of the builder's dictionary.</summary>
```
Comment (rpetrusha) [inline xml/System.Collections.Immutable/ImmutableSortedDictionary.xml]:
```suggestion
        <summary>Creates an immutable sorted dictionary from the current contents of the builder's dictionary.</summary>
```

File: xml/System.Collections.Immutable/ImmutableSortedSet.xml [hunk 6]
```diff
@@ -433,10 +433,10 @@
         <Parameter Name="builder" Type="System.Collections.Immutable.ImmutableSortedSet&lt;TSource&gt;+Builder" RefType="this" Index="0" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <typeparam name="TSource">To be added.</typeparam>
-        <param name="builder">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="TSource">The type of the elements in the immutable sorted set.</typeparam>
+        <param name="builder">The builder to create the immutable sorted set from.</param>
+        <summary>Creates an immutable sorted set of the current contents of the builder's set.</summary>
```
Comment (rpetrusha) [inline xml/System.Collections.Immutable/ImmutableSortedSet.xml]:
```suggestion
        <summary>Creates an immutable sorted set from the current contents of the builder's set.</summary>
```

**Comment by Jozkee** [comment]
@rpetrusha thanks for the comments, I've addressed them. can we merge this?

**Comment by rpetrusha** [comment]
Yes, I'll merge it now, @Jozkee. Thanks for the additional changes.

**Pull request merged**